Ginger/GingerCoreNET/Application Models/Delta/APIDelta/APIDeltaUtils.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace GingerCoreNET.Application_Models
+{
+    public class APIDeltaUtils
+    {
+        public static void ComparisonUtility(ObservableList<ApplicationAPIModel> learnedAPIModelsList, ObservableList<DeltaAPIModel> deltaAPIModelsList)
+        {
+            ObservableList<ApplicationAPIModel> existingAPIModelsList = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<ApplicationAPIModel>();
+
+            //for (int i = 0; i < learnedAPIModelsList.Count; i++)
+            foreach (ApplicationAPIModel apiModelLearned in learnedAPIModelsList)
+            {
+                //ApplicationAPIModel apiModelLearned = learnedAPIModelsList[i];
+
+                DeltaAPIModel apiModelDelta = new DeltaAPIModel();
+                apiModelDelta.learnedAPI = apiModelLearned;
+                List<ApplicationAPIModel> matchingAPIModels;
+
+                if (apiModelLearned.APIType == ApplicationAPIUtils.eWebApiType.SOAP)
+                    matchingAPIModels = existingAPIModelsList.Where(m => m.EndpointURL.Equals(apiModelLearned.EndpointURL, StringComparison.OrdinalIgnoreCase) && m.APIType == apiModelLearned.APIType && m.SOAPAction.Equals(apiModelLearned.SOAPAction)).ToList();
+                else
+                    matchingAPIModels = existingAPIModelsList.Where(m => m.EndpointURL.Equals(apiModelLearned.EndpointURL, StringComparison.OrdinalIgnoreCase) && m.APIType == apiModelLearned.APIType && m.RequestType == apiModelLearned.RequestType).ToList();
+
+                if (matchingAPIModels != null && matchingAPIModels.Count > 0)
+                {
+                    ApplicationAPIModel matchingModel = matchingAPIModels[0];
+                    matchingAPIModels = CompareAPIModels(apiModelLearned, matchingAPIModels);
+                    if (matchingAPIModels.Count > 0)
+                    {
+                        apiModelDelta.comparisonStatus = DeltaAPIModel.eComparisonOutput.Unchanged;         // UNCHANGED
+                        apiModelDelta.matchingAPIModel = matchingAPIModels[0];
+                        apiModelDelta.IsSelected = false;
+                    }
+                    else
+                    {
+                        apiModelDelta.comparisonStatus = DeltaAPIModel.eComparisonOutput.Modified;          // MODIFIED
+                        apiModelDelta.matchingAPIModel = matchingModel;
+                        apiModelDelta.IsSelected = true;
+                    }
+                }
+                else
+                {
+                    apiModelDelta.comparisonStatus = DeltaAPIModel.eComparisonOutput.New;                   // NEW API
+                    apiModelDelta.IsSelected = true;
+                }
+
+                deltaAPIModelsList.Add(apiModelDelta);
+            }
+        }
+
+        public static ApplicationAPIModel CreateAPIModelObject(ApplicationAPIModel sourceAPIModel)
```

Comment:
this is not needed anymore, no?

Reply:
This is needed for the case where user selects either Existing or Learned API as Merged API